flutter_local_notifications/example/lib/main.dart

```diff
@@ -499,22 +505,51 @@ class _HomePageState extends State<HomePage> {
     var androidPlatformChannelSpecifics = AndroidNotificationDetails(
         'your channel id', 'your channel name', 'your channel description',
         importance: Importance.max, priority: Priority.high, ticker: 'ticker');
-    var platformChannelSpecifics =
-        NotificationDetails(android: androidPlatformChannelSpecifics);
-    await flutterLocalNotificationsPlugin.show(
-        0, 'plain title', 'plain body', platformChannelSpecifics,
+    var platformChannelSpecifics = NotificationDetails(android: androidPlatformChannelSpecifics);
+    await flutterLocalNotificationsPlugin.show(0, 'plain title', 'plain body', platformChannelSpecifics,
         payload: 'item x');
   }
 
+  Future<void> _showFullScreenNotification() async {
+    await showDialog(
+        context: context,
+        child: AlertDialog(
+          title: Text('Turn off your screen'),
+          content: Text('to see the full-screen intent in 3 seconds, press OK and TURN OFF your screen'),
+          actions: <Widget>[
+            FlatButton(
+              child: Text('Cancel'),
+              onPressed: () {
+                Navigator.pop(context);
+              },
+            ),
+            FlatButton(
+              child: Text('OK'),
+              onPressed: () async {
+                Navigator.pop(context);
```

Comment:
Popping first and then doing the work after it is a bit counter-intuitive. Can you instead use `zonedSchedule` to schedule a full-screen intent notification to take place in 3 seconds and pop after scheduling is done? That would help to guarantee the notification runs regardless of what state the app is in